src/backends/interflop-mca/interflop_mca.c

```diff
@@ -122,135 +152,85 @@ static double _mca_rand(void) {
   return tinymt64_generate_doubleOO(&random_state);
 }
 
-static inline double pow2d(int exp) {
-  double res = 0;
-  uint64_t x[1];
-  // specials
-  if (exp == 0)
-    return 1;
-
-  if (exp > 1023) { /*exceed max exponent*/
-    *x = DOUBLE_PLUS_INF;
-    res = *((double *)x);
-    return res;
-  }
-  if (exp < -1022) { /*subnormal*/
-    *x = ((uint64_t)DOUBLE_PMAN_MSB) >> -(exp + DOUBLE_EXP_MAX);
-    res = *((double *)x);
-    return res;
-  }
+static inline double pow2d(int exp) { return ldexp(1.0, exp); }
```

Comment:
This is old. But I kind of remember we didn't use existing function because our very specific case allows for some short cuts and performance gain. Did you check the parformance of your backend compare to the original quad one?

Reply:
Yes, you're right. Actually, as for `qnoise` we don't need to check specials cases.